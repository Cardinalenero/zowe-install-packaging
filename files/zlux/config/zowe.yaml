components:
  app-server:
    node:
      https:
        ipAddresses:
        - 0.0.0.0
        port: 7556
        keys:
        - "../defaults/serverConfig/zlux.keystore.key"
        certificates:
        - "../defaults/serverConfig/zlux.keystore.cer"
        certificateAuthorities:
        - "../defaults/serverConfig/apiml-localca.cer"
      mediationLayer:
        server:
          hostname: localhost
          port: 7554
          isHttps: true
        enabled: false
      headers:
        X-frame-Options:
          override: true
          value: sameorigin
    agent:
      host: localhost
      jwt:
        fallback: true
      https:
        ipAddresses:
        - 127.0.0.1
        port: 7557
      mediationLayer:
        serviceName: zss
        enabled: false
    privilegedServerName: ZWESIS_STD
    productDir: "../defaults"
    siteDir: "../deploy/site"
    instanceDir: "../deploy/instance"
    groupsDir: "../deploy/instance/groups"
    usersDir: "../deploy/instance/users"
    pluginsDir: "../defaults/plugins"
    dataserviceAuthentication:
      defaultAuthentication: saf
      rbac: false
      implementationDefaults:
        saf:
          plugins:
          - org.zowe.zlux.auth.safsso
  zss:
    agent:
      host: localhost
      jwt:
        fallback: true
      https:
        ipAddresses:
        - 127.0.0.1
        port: 7557
      mediationLayer:
        serviceName: zss
        enabled: false

    