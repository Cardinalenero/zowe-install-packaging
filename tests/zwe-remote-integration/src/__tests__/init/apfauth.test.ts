/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */

import { REMOTE_SYSTEM_INFO, TEST_COLLECT_SPOOL } from '../../config/TestConfig';
import ZoweYamlType from '../../config/ZoweYamlType';
import { RemoteTestRunner } from '../../zos/RemoteTestRunner';
import { ZoweYaml } from '../../config/ZoweYaml';
import { FileType, TestFileActions, TestFile } from '../../zos/TestFileActions';

const testSuiteName = 'init-apfauth';
describe(`${testSuiteName}`, () => {
  let testRunner: RemoteTestRunner;
  let cfgYaml: ZoweYamlType;
  let cleanupDatasets: TestFile[] = []; // a list of datasets deleted after every test

  beforeAll(() => {
    testRunner = new RemoteTestRunner(testSuiteName);
  });
  beforeEach(() => {
    cfgYaml = ZoweYaml.basicZoweYaml();
  });

  afterEach(async () => {
    if (TEST_COLLECT_SPOOL) {
      await testRunner.collectSpool();
    }
    // re-created in every `init` subcommand based on changes to zowe yaml command...
    const jcllib: TestFile = { name: REMOTE_SYSTEM_INFO.jcllib, type: FileType.DS_NON_CLUSTER };

    // try to delete everything we know about
    await TestFileActions.deleteAll([...cleanupDatasets, jcllib]);
    cleanupDatasets = [];
  });

  describe('(LONG)', () => {});

  describe('(SHORT)', () => {
    it('apf disable cfgmgr', async () => {
      cfgYaml.zowe.useConfigmgr = false;
      const result = await testRunner.runZweTest(cfgYaml, 'init apfauth');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot();
      expect(result.rc).toBe(60); // 60 is expected error code...
    });

    it('apf bad ds prefix', async () => {
      cfgYaml.zowe.setup.dataset.prefix = 'ZOWEAD6.ZWETEST.NOEXIST';
      const result = await testRunner.runZweTest(cfgYaml, 'init apfauth --dry-run');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot();
      expect(result.rc).toBe(231);
    });

    it('apf dry-run', async () => {
      const result = await testRunner.runZweTest(cfgYaml, 'init apfauth --dry-run');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot();
      expect(result.rc).toBe(231);
    }, 400000);

    /* it('apf simple --dry-run', async () => {
      const result = await testRunner.runZweTest(cfgYaml, 'init apfauth --dry-run');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot();
      expect(result.rc).toBe(0); // 60 is expected...
    });

    /* it('apf security-dry-run', async () => {
      const result = await testRunner.runZweTest(cfgYaml, 'init apfauth --security-dry-run');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot();
      expect(result.rc).toBe(0); // 60 is expected...  });
    });*/
  });
});
