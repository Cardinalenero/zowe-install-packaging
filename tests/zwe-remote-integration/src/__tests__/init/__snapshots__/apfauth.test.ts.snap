// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init-apfauth (SHORT) apf bad ds prefix 1`] = `
"
$ Temporary directory '/tmp/.zweenv-0000' created.
Zowe will remove it on success, but if zwe exits with a non-zero code manual cleanup would be needed.
-------------------------------------------------------------------------------
>> APF authorize load libraries

cp: FSUM6258 cannot open file "//'TEST.DATASET.PFX.NOEXIST.SZWESAMP(ZWEGENER)'": EDC5049I The specified file name could not be located. (errno2=0xC00B0641)
ERROR: ZWEL0143E Cannot find data set member 'TEST.DATASET.PFX.NOEXIST.SZWESAMP(ZWEGENER)'. You may need to re-run zwe install."
`;

exports[`init-apfauth (SHORT) apf disable cfgmgr 1`] = `
"
$ Error ZWEL0316E: Command requires zowe.useConfigmgr=true to use."
`;
